base/server/src/main/java/org/dogtagpki/server/PKIClientSocketListener.java

```diff
@@ -174,9 +178,35 @@ public void alertSent(SSLAlertEvent event) {
     public void handshakeCompleted(SSLHandshakeCompletedEvent event) {
         String method = "PKIClientSocketListener.handshakeCompleted: ";
         logger.debug(method + "begins");
+
+	boolean invalidateAfterHandshake = false;
+
+	CMSEngine cms = CMS.getCMSEngine();
+
+        EngineConfig cfg =  null ;
+        JssSubsystemConfig jcfg = null;
+
+        if(cms != null) {
+            cfg = cms.getConfig();
+            if(cfg != null) {
+                jcfg = cfg.getJssSubsystemConfig();
+                if(jcfg != null) {
+                    try {
+                        invalidateAfterHandshake = jcfg.getBoolean("ssl.client.invalidateSessionAfterHandshake",false);
+                    } catch(EBaseException e) {
+                        invalidateAfterHandshake = false;
+                    }
+                }
+            }
+        }
+	logger.debug("PKIClientSocketListener: invalidateAfterHandshake = " + invalidateAfterHandshake);
+
         try {
             SSLSocket socket = event.getSocket();
 
+	    if(socket != null && invalidateAfterHandshake) {
```

Comment:
again, put a comment that say something like "// invalidateAfterHandshake is for testing only; do not use